designer/client/src/components/graph/node-modal/tests/ExpressionTestResults.tsx

```diff
@@ -14,6 +15,14 @@ interface ExpressionTestResultsProps {
 export default function ExpressionTestResults(props: PropsWithChildren<ExpressionTestResultsProps>): JSX.Element {
     const { fieldName, resultsToShow } = props;
     const [hideTestResults, toggleTestResults] = useState(false);
+    const [fitsMaxHeight, setFitsMaxHeight] = useState(true);
+    const testValueRef: React.Ref<HTMLTextAreaElement> = useRef(null);
+    useEffect(() => {
+        if (testValueRef.current && testValueRef.current.scrollHeight > HiddenTextareaPixelHeight) {
+            toggleTestResults((s) => !s);
+            setFitsMaxHeight(false);
+        }
+    }, []);
```

Comment:
As I see it's not fully working solution, so my proposal is:

Rename

```
const [hideTestResults, toggleTestResults] = useState(true);
```
to
```
const [collapsedTestResults, setCollapsedTestResults] = useState(true);
```

Since we only want to display the field grey out when content is higher than 100 we should change

```
shouldHideTestResults={hideTestResults}
```

to

```
shouldHideTestResults={collapsedTestResults && !fitsMaxHeight}
```

Reply:
I didn't know it's advised to avoid useEffects in React, thank you so much for the insight!
I removed the useEffect, changed variables and the hide result check.